README.md

```diff
@@ -1,2 +1,454 @@
+
+With the exception of rights amplification (below), each delegation of a capability MUST have equal or lesser power from its witness. The time bounds MUST also be equal to or contained inside the time bounds of the witnesses time bounds. This lowering of rights at each delegation is called "attenuation".
+
+## 5.4 Rights Amplification
+
+Some capabilities are more than the sum of their parts. The canonical example is a can of soup and a can opener. You need both to access the soup inside the can, but the can opener may come from a completely separate source than the can of soup. Such semantics MAY be implemented in UCAN capabilities. This means that validating a particular capabilities MAY require more than one direct witness. The relevant witnesses MAY be of a different resource and action from the amplified capability. The delegated capability MUST have this behaviour in its semantics, even if the witnesses do not.
+
+## 5.6 Content Identifiers
+
+UCANs MAY be referenced by content ID (CID), per the [multiformats/cid](https://github.com/multiformats/cid) specification. The resolution of these addresses is left to the implementation and end user, and MAY (non-exclusively) include the following: local store, distributed hash table (DHT), gossip network, or RESTful service.
+
+CIDs MAY be used to refer to any UCAN: a witness in a delegation chain, or an entire UCAN. This has many benefits, some of which are outlined in the implementation recommendations of this document.
+
+Due to the potential for unresolvable CIDs, this SHOULD NOT be the preferred method of transmission. "Inline witnesses" SHOULD be used whenever possible, and complete UCANs SHOULD be expanded. When a CID is used, it is RECOMMENDED that it be substituted as close to the top UCAN as possible (i.e. the invocation), and as few witnesses be referenced by CID, to keep the number of required CID resolutions to a minimum. As UCANs are signed, all further delegations would require CID resolution, and so SHOULD NOT be used when the intention is delegation rather than invocation. 
+
+## 5.7 Revocation
+
+Any issuer of a UCAN MAY later revoke that UCAN, or the capabilities that have been derived from it further downstream in a UCAN chain.
+
+This mechanism is eventually consistent, and SHOULD be considered a last line of defence against abuse. Proactive expiry via time bounds or other constraints SHOULD be preferred, as they do not require learning more information than what would be available on an offline computer.
+
+While some resources are centralized (e.g. access to a server), and others are unbound from specific locations (e.g. a CRDT), and will take longer for the revocation to propagate.
```

Comment:
> While some resources are centralized (e.g. access to a server), ~and~ others are unbound from specific locations (e.g. a CRDT) ~,~ and will take longer for the revocation to propagate.

The "and" and comma are not needed in this sentence.

Reply:
@bgins are you not a fan of the Oxford comma?

Reply:
![](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fwww.collegemagazine.com%2Fwp-content%2Fuploads%2F2018%2F07%2F32202880_754324651432406_3849994953123954688_n.jpg&f=1&nofb=1)

Reply:
Haha, that's a great comic 😁 

I'm alright with Oxford commas. I might be parsing this sentence incorrectly, but it reads to me like a while clause followed by clause with two items. I've typically seen the Oxford comma in lists with at least three items.